package/MDAnalysis/analysis/align.py

```diff
@@ -359,29 +424,175 @@ def alignto(mobile, reference, select="all", mass_weighted=False,
     ref_coordinates = ref_atoms.positions - ref_com
     mobile_coordinates = mobile_atoms.positions - mobile_com
 
-    old_rmsd = rms.rmsd(mobile_coordinates, ref_coordinates)
-
-    R, new_rmsd = rotation_matrix(mobile_coordinates, ref_coordinates, weights=weights)
+    old_rmsd = rms.rmsd(mobile_coordinates, ref_coordinates, weights)
 
     if subselection is None:
-        atoms = mobile.universe.atoms
-    elif type(subselection) is str:
-        atoms = mobile.select_atoms(subselection)
+        # mobile_atoms is Universe
+        mobile_atoms = mobile.universe.atoms
+    elif isinstance(subselection, str):
+        # select mobile_atoms from string
+        mobile_atoms = mobile.select_atoms(subselection)
     else:
         try:
-            atoms = subselection.atoms
+            # treat subselection as AtomGroup
+            mobile_atoms = subselection.atoms
         except AttributeError:
-            raise TypeError("subselection must be a selection string, a AtomGroup or Universe or None")
-
-    atoms.translate(-mobile_com)
-    atoms.rotate(R)
-    atoms.translate(ref_com)
+            raise TypeError("subselection must be a selection string, an"
+                            " AtomGroup or Universe or None")
 
+    # _fit_to DOES subtract center of mass, will provide proper min_rmsd
+    mobile_atoms, new_rmsd = _fit_to(mobile_coordinates, ref_coordinates,
+                                     mobile_atoms, mobile_com, ref_com,
+                                     weights=weights)
     return old_rmsd, new_rmsd
 
 
-def rms_fit_trj(traj, reference, select='all', filename=None, rmsdfile=None, prefix='rmsfit_',
-                mass_weighted=False, tol_mass=0.1, strict=False, force=True, quiet=False, **kwargs):
+class AlignTraj(AnalysisBase):
+    """rms_align trajectory to a reference structure using a selection.
```

Comment:
I'll put it on my todo list for tomorrow. Thanks for the tremendous help,
I'm sure it can get exhausting looking over this after doing all your own
work.
On Thu, Jun 2, 2016 at 12:32 AM kain88-de <notifications@github.com> wrote:

> In package/MDAnalysis/analysis/align.py
> <https://github.com/MDAnalysis/mdanalysis/pull/856#discussion_r65495624>:
>
> >      return old_rmsd, new_rmsd
> >
> >
> > -def rms_fit_trj(traj, reference, select='all', filename=None, rmsdfile=None, prefix='rmsfit_',
> > -                mass_weighted=False, tol_mass=0.1, strict=False, force=True, quiet=False, **kwargs):
> > +class AlignTraj(AnalysisBase):
> > +    """rms_align trajectory to a reference structure using a selection.
>
> OK one last issue. We generate new class variables like self.rmsd or
> self.weights during a run. All that are now hidden to the user with a _
> prefix should be documented as well. I know we didn't to it for other
> classes but I think we should get in the habit of that for analysis
> classes. @jdetle <https://github.com/jdetle> can you look up in the numpy
> doc-style docs what the recommended way for exposing these docs is?
>
> *Note* every member variables that you don't want to document should get
> a _ prefix. writer is a good example for that.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/MDAnalysis/mdanalysis/pull/856/files/0596f596f04c9fd05b5768453423a5d3b4431a15#r65495624>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AKcARkZGO0NWBTS-gPkzeeZEVMZp5DwTks5qHocpgaJpZM4Ig_Re>
> .
>
-- 
Have a wonderful day,
John J. Detlefs